src/history/hs_cursor.c

```diff
@@ -373,24 +378,22 @@ __hs_find_upd_int(WT_SESSION_IMPL *session, uint32_t btree_id, WT_ITEM *key,
                 break;
             }
 
-            /*
-             * If we find a history store record that either corresponds to the on-disk value or is
-             * newer than it then we should use the on-disk value as the base value and apply our
-             * modifies on top of it.
-             */
-            if (on_disk_tw->start_ts < hs_start_ts_tmp ||
-              (on_disk_tw->start_ts == hs_start_ts_tmp &&
-                on_disk_tw->start_txn <= hs_cbt->upd_value->tw.start_txn)) {
-                /* Fallback to the onpage value as the base value. */
-                orig_hs_value_buf = hs_value;
-                hs_value = on_disk_buf;
-                upd_type = WT_UPDATE_STANDARD;
-                break;
-            }
-
             WT_ERR(hs_cursor->get_value(hs_cursor, &hs_stop_durable_ts_tmp, &durable_timestamp_tmp,
               &upd_type_full, hs_value));
             upd_type = (uint8_t)upd_type_full;
+
+            /*
+             * The current onpage value in memory may also be in the history store if checkpoint
```

Comment:
Hmm, is the onpage value invalid to read at this point? I'm finding this comment a bit confusing.

Reply:
Yes, it is an removed overflow value removed during the checkpoint reconciliation.

Reply:
Hmm, what about a reader that reads the onpage value directly. Is that a problem? Or is it only because we took a pointer to the previous onpage value.

Reply:
We have already appended the onpage value to the update chain in this case during checkpoint reconciliation so a reader reading that version is OK.

Reply:
Ah right, that makes sense. Thanks for explaining @quchenhao.